### ISSUE TITLE
native-image should benefit of VarHandle

### ISSUE THREAD

**Issue by franz1981** [netty/netty#15827]
Thanks to #15764 now it should be possible for `native-image` to leverage `VarHandle` powers, which can give some interesting performance uplift on `ByteBuf` accesses, because the accessibility checks won't involve `volatile` loads which could lead to 

It would be nice, then, to verify if  is possible change
https://github.com/netty/netty/blob/2bf7e62254501c0849b0df4bb8999455b687c8c6/common/src/main/java/io/netty/util/internal/PlatformDependent.java#L268
in order to remove the limit of using `VarHandle`.

The risks are that other cases than the reference count ones, would break e.g. `byteArrayView`/`byteBufferView` and need to verify such.
But, if both mentioned cases won't fail this can give a very nice performance advantage there too, in a new `no Unsafe` world - to `native-image` as well.

**Comment by franz1981** [comment]
@zakkak as well ;)
